### PR TITLE
Add evaluated_keys to missing aws terraform checks

### PR DESCRIPTION
**Pull request by YaaraVerner**
Add evaluated_keys to missing aws terraform checks.
Refactor some checks

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by gruebel — COMMENTED** [review]
nice work

File: checkov/terraform/checks/resource/aws/AutoScalingTagging.py [hunk 1]
```diff
@@ -14,9 +15,12 @@ def scan_resource_conf(self, conf):
...
             Looks for tag or tags
         """
-
         if "tag" in conf.keys() or "tags" in conf.keys():
             return CheckResult.PASSED
-        return  CheckResult.FAILED
+        return CheckResult.FAILED
+
+    def get_evaluated_keys(self) -> List[str]:
+        return ['tag', 'tags']
```
Comment (nimrodkor) [inline checkov/terraform/checks/resource/aws/AutoScalingTagging.py:23]: @YaaraVerner Isn't it either-or? Either `tag` was used, or `tags`, right?

Comment (YaaraVerner) [reply checkov/terraform/checks/resource/aws/AutoScalingTagging.py:23]: it is. but the key doesn't exist it ignored

File: checkov/terraform/checks/resource/aws/S3AllowsAnyPrincipal.py [hunk 2]
```diff
@@ -14,29 +15,29 @@ def __init__(self):
...
-                            if principal == '*':
-                                return CheckResult.FAILED
-                            elif 'AWS' in statement['Principal']:
-                                # Can be a string or an array of strings
-                                aws = statement['Principal']['AWS']
-                                if (type(aws) == str and aws == '*') or (type(aws) == list and '*' in aws):
-                                    return CheckResult.FAILED
-                except: # nosec
-                    pass
+        if 'policy' not in conf.keys() or not isinstance(conf['policy'][0], str):
```
Comment (tronxd) [inline checkov/terraform/checks/resource/aws/S3AllowsAnyPrincipal.py:18]: How does it handle the case where `conf['policy']` is `None` or an empty list?

Comment (YaaraVerner) [reply checkov/terraform/checks/resource/aws/S3AllowsAnyPrincipal.py:18]: is it possible? I just contained the first two if conditions

**Review by tronxd — APPROVED** [review]
Nice!

**Pull request merged**